New general sidebar article to help devs choose between PWA and extensions (#2729)
MicrosoftDocs/edge-developer: microsoft-edge/web-platform/sidebar.md

author: MSEdgeTeam
ms.author: msedgedevrel
ms.topic: conceptual
ms.prod: microsoft-edge
ms.date: 07/24/2023
---
# Develop for the sidebar

The sidebar in Microsoft Edge is a persistent pane located on the side of the browser, which coexists with the primary content of the browser. The sidebar allows users to easily access popular websites and utilities alongside their browser tabs. The content in the sidebar augments the user's primary task by enabling side-by-side browsing and minimizing the need to switch contexts between browser tabs. With the sidebar in Microsoft Edge, users can access the productivity tools they need while staying in their workflow.

![The sidebar in Microsoft Edge, showing useful tools, alongside the main tab](./sidebar-images/sidebar.png)

As a developer, you can leverage the sidebar for your own experiences in two ways:

* By adapting your existing Progressive Web App (PWA) to run in the sidebar.
* By making use of the Sidebar API in your Microsoft Edge extension.

<!-- ====================================================================== -->
## Choose the right approach

> The two different ways in which you can leverage the sidebar are designed to meet different developer needs. The following table summarizes the key differences between the two approaches:

Reviewer: ```suggestion
The two different ways in which you can leverage the sidebar are designed to meet different developer needs. The following are the key differences between sidebar apps and sidebar extensions.
```
* Focus on the technical content rather than documentation constructs ("the following table").
* Name the specific approaches.

|Approach | Sidebar app | Sidebar extension |
|---|---|---|
| Definition | A Progressive Web App (PWA) that runs in the sidebar. | A Microsoft Edge extension that displays content in the sidebar. |
| Distribution | Like a website, the app is hosted on the app's web server. | The extension is distributed through Microsoft Edge Add-ons. |
| Capabilities | The app can make use of all web APIs and features, as well as PWA capabilities such as offline support, push, or file system access. | The extension can make use of powerful extension APIs, such as reading or modifying the current page. |

> | Context | The app runs in its own web context, separate from the main tab. | The extension can access the current page and modify it via its content script. |

Reviewer: ```suggestion
| Context | The app runs in its own web context, separate from the main tab. | The extension can access the current webpage and modify the webpage via the extension's content script. |
```
* Prefer direct wording instead of "it" and "its" indirection pointers.
* Tech review "its": Is it the extension's content script, or the webpage's content script?

> | Installation | The user installs the app by interacting with the sidebar when visiting the app's website. | The extension is installed by the user from Microsoft Edge Add-ons. |

Reviewer: ```suggestion
| Installation | The user installs the app by interacting with the sidebar when visiting the app's website. | The extension is installed by the user from the Microsoft Edge Add-ons website. |
```
* What's meant by "user from Microsoft Edge Add-ons" - a website, or a UI?  If UI, add hint words about where in UI.

| User experience | The app has its own icon in the sidebar. Clicking the icon opens the sidebar if it was closed and displays the app's contents. | The user experience depends on the extension. The extension can open the sidebar depending on the websites the user visits in the browser. |
| Knowledge required | Web development and PWA knowledge. | Extension development knowledge. |

<!-- ====================================================================== -->
## Sidebar apps

Sidebar apps allow you to adapt your existing Progressive Web App (PWA) to securely run alongside other browser tabs for a side-by-side co-browsing experience, which is a great choice for web developers who want to offer companion experiences including social, messaging, or media apps that don't require advanced extensions capabilities.

> We plan to help users discover and install sidebar apps as they browse the web, providing great discoverability for your app with just a simple change to your web application manifest file. This is a great option if you want to reuse your existing web app for the sidebar in Microsoft Edge and make full use of the web capabilities that are available to PWAs.

Reviewer: ```suggestion
Sidebar apps plan to help users discover and install sidebar apps as they browse the web, providing great discoverability for your app with just a simple change to your web application manifest file. This is a great option if you want to reuse your existing web app for the sidebar in Microsoft Edge and make full use of the web capabilities that are available to PWAs.
```
* usually try to avoid "we", per Style Guides.

To learn more about PWAs and how to create them, see [Overview of Progressive Web Apps (PWAs)](../progressive-web-apps-chromium/index.md). And to learn more about adapting your existing PWA to the sidebar in Microsoft Edge, see [Build PWAs for the sidebar in Microsoft Edge](../progressive-web-apps-chromium/how-to/sidebar.md).

<!-- ====================================================================== -->
## Sidebar extensions

Microsoft Edge extensions can optionally use the sidebar API to show a custom UI in the sidebar in Microsoft Edge. This is in addition to the other places extensions also appear in, such as in the Microsoft Edge toolbar, or as popups.

> If your experience relies on powerful extensions APIs, for example to read or modify the current page, then a sidebar extension provides these capabilities. Users can find and install these extensions through Microsoft Edge Add-ons. We are continuing to evolve these sidebar extensibility  models and welcome developer feedback at [explainer link?]

Reviewer: ```suggestion
If your experience relies on powerful extensions APIs, for example to read or modify the current page, then a sidebar extension provides these capabilities. Users can find and install these extensions through Microsoft Edge Add-ons. We are continuing to evolve these sidebar extensibility models and welcome developer feedback at [explainer link?]
```
* tighten space
* needs link

---
title: Develop for the sidebar
description: Learn to develop experiences for the sidebar in Microsoft Edge, by either installing a Progressive Web App, or using the extension sidebar API.
author: MSEdgeTeam
ms.author: msedgedevrel
ms.topic: conceptual
ms.prod: microsoft-edge
ms.date: 07/24/2023
---
# Develop for the sidebar

> The sidebar in Microsoft Edge is a persistent pane located on the side of the browser, which coexists with the primary content of the browser. The sidebar allows users to easily access popular websites and utilities alongside their browser tabs. The content in the sidebar augments the user's primary task by enabling side-by-side browsing and minimizing the need to switch contexts between browser tabs. With the sidebar in Microsoft Edge, users can access the productivity tools they need while staying in their workflow.

Reviewer: ```suggestion
The sidebar in Microsoft Edge is a persistent pane located on the side of the browser, which coexists with the primary content of the browser. The sidebar allows users to easily access popular websites and utilities alongside their browser tabs. The content in the sidebar augments the user's primary task by enabling side-by-side browsing and minimizing the need to switch contexts between browser tabs. By using the sidebar in Microsoft Edge, users can access the productivity tools they need while staying in their workflow.
```
https://learn.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/u/using-by-using-with - "Don't use _with_ as a synonym for _using_ or _by using_."
Alts:
* By using
* Using

Reviewer: incorp'd

![The sidebar in Microsoft Edge, showing useful tools, alongside the main tab](./sidebar-images/sidebar.png)

As a developer, you can leverage the sidebar for your own experiences in two ways:

> * By adapting your existing Progressive Web App (PWA) to run in the sidebar.

Reviewer: ```suggestion
* By adapting your existing Progressive Web App (PWA) to run partly or entirely in the sidebar.
```
tech review this change

Reviewer: i'm not incorp'g w/o tech review, might close

* By making use of the Sidebar API in your Microsoft Edge extension.

<!-- ====================================================================== -->
## Choose the right approach

The two different ways in which you can leverage the sidebar are designed to meet different developer needs. The following are the key differences between sidebar apps and sidebar extensions:

> |Approach | Sidebar app | Sidebar extension |

Reviewer: ```suggestion
| Aspect | Sidebar app | Sidebar extension |
```
instead of pointing to the right, point down.  What's below in this column isn't 7 approaches, but 7 aspects of the two approaches.

| Context | The app runs in its own web context, separate from the main tab. | The extension can access the current webpage and modify the webpage via the extension's content script. |
| Installation | The user installs the app by interacting with the sidebar when visiting the app's website. | The extension is installed by the user from the [Microsoft Edge Add-ons website](https://microsoftedge.microsoft.com/addons/Microsoft-Edge-Extensions-Home). |
| User experience | The app has its own icon in the sidebar. Clicking the icon opens the sidebar if it was closed and displays the app's contents. | The user experience depends on the extension. The extension can open the sidebar depending on the websites the user visits in the browser. |
| Technologies used | Web development and PWA. See [Overview of Progressive Web Apps (PWAs)](../progressive-web-apps-chromium/index.md). | Extension development.  See [Overview of Microsoft Edge extensions](../extensions-chromium/index.md). |

<!-- ====================================================================== -->
## Sidebar apps

> Sidebar apps allow you to adapt your existing Progressive Web App (PWA) to securely run alongside other browser tabs for a side-by-side co-browsing experience, which is a great choice for web developers who want to offer companion experiences including social, messaging, or media apps that don't require advanced extensions capabilities.

Reviewer: ```suggestion
Sidebar apps allow you to adapt your existing Progressive Web App (PWA) to securely run alongside other browser tabs for a side-by-side co-browsing experience. This approach is a great choice for web developers who want to offer companion experiences including social, messaging, or media apps that don't require advanced extensions capabilities.
```
Acrolinx might suggest such breaking up sentence.  readability = short words/ phrases/ sentences/ paragraphs/ sections.

Reviewer: incorp'd

> The sidebar app infrastructure helps users discover and install sidebar apps as they browse the web, providing great discoverability for your app with just a simple change to your web application manifest file. Sidebar apps are a great option if you want to reuse your existing web app for the sidebar in Microsoft Edge and make full use of the web capabilities that are available to PWAs.

Reviewer: ```suggestion
The sidebar app infrastructure helps users discover and install sidebar apps as they browse the web, providing great discoverability for your app just by making a simple change to your web application manifest file. Sidebar apps are a great option if you want to reuse your existing web app for the sidebar in Microsoft Edge and make full use of the web capabilities that are available to PWAs.
```
an alternative to consider

fwiw:
https://learn.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/u/using-by-using-with - "Don't use with as a synonym for using or by using."

Reviewer: incorp'd

To learn more about PWAs and how to create them, see [Overview of Progressive Web Apps (PWAs)](../progressive-web-apps-chromium/index.md). And to learn more about adapting your existing PWA to the sidebar in Microsoft Edge, see [Build PWAs for the sidebar in Microsoft Edge](../progressive-web-apps-chromium/how-to/sidebar.md).

<!-- ====================================================================== -->
## Sidebar extensions

Microsoft Edge extensions can optionally use the sidebar API to show a custom UI in the sidebar in Microsoft Edge. This is in addition to the other places extensions also appear in, such as in the Microsoft Edge toolbar, or as popups.

> If your experience relies on powerful extensions APIs, for example to read or modify the current page, then a sidebar extension provides these capabilities. Users can find and install these extensions through the [Microsoft Edge Add-ons website](https://microsoftedge.microsoft.com/addons/Microsoft-Edge-Extensions-Home). The Microsoft Edge team continues to evolve these sidebar extensibility models and welcomes your feedback. To leave feedback, [create a new issue](https://github.com/MicrosoftEdge/MSEdgeExplainers/issues/new/) on the repo or [search for existing issues](https://github.com/MicrosoftEdge/MSEdgeExplainers/issues) and join existing conversations.

Reviewer: ```suggestion
If your app's experience relies on powerful extensions APIs, for example to read or modify the current page, then a sidebar extension provides these capabilities. Users can find and install these extensions through the [Microsoft Edge Add-ons website](https://microsoftedge.microsoft.com/addons/Microsoft-Edge-Extensions-Home). The Microsoft Edge team continues to evolve these sidebar extensibility models and welcomes your feedback. To leave feedback, [create a new issue](https://github.com/MicrosoftEdge/MSEdgeExplainers/issues/new/) on the repo or [search for existing issues](https://github.com/MicrosoftEdge/MSEdgeExplainers/issues) and join existing conversations.
```
alts:
If the experience that your app provides
If the user experience that your app provides

Reviewer: incorp'd
